### PR TITLE
enhancement(external docs): Multiple condition types

### PR DESCRIPTION
**Pull request by lucperkins**
Affected sections:

https://deploy-preview-10157--vector-project.netlify.app/docs/reference/configuration/transforms/filter/#condition
https://deploy-preview-10157--vector-project.netlify.app/docs/reference/configuration/transforms/sample/#exclude
https://deploy-preview-10157--vector-project.netlify.app/docs/reference/configuration/transforms/throttle/#exclude

I don't really love this just yet but it's a decent first stab. Curious what others think about this approach.

### CODE REVIEW

File: website/cue/reference/components/transforms/filter.cue [hunk 1]
```diff
@@ -27,15 +27,17 @@ components: transforms: filter: {
 	configuration: {
 		condition: {
 			description: """
-				The condition to be matched against every input event. Only messages that pass the condition will
-				be forwarded.
+				The condition to be matched against every input event. Only messages that pass the condition are
+				forwarded; messages that don't pass are dropped.
```
Comment (jszwedko) [inline website/cue/reference/components/transforms/filter.cue:31]: Maybe we could just add a note in prose here in the description that, in addition to the `type`/`source` map below, there is a short-hand of just passing a VRL string condition? I realize that sort of configuration field type union is difficult to model.

**Review by jszwedko — COMMENTED** [review]
Nice! I was actually just lamenting the lack of documentation around conditions again today so this is very timely 😄 

The rendered output looks good to me. However, anywhere that we have a `condition` should support both `vrl` and `datadog_search` (also `check_fields` but that is deprecated) so you could probably drop the configuration of syntaxes at the component level and just render them all.

**Comment by lucperkins** [comment]
@jszwedko I was under the impression, I believe from a comment by @leebenson, that `datadog_search` is _not_ always available. Lee, could you weigh in on this here?

File: website/cue/reference/components/transforms/route.cue [hunk 2]
```diff
@@ -42,12 +42,7 @@ components: transforms: route: {
 							condition will be included in this route.
 							"""
 						required: true
-						type: string: {
-							examples: [
-								#".status_code != 200 && !includes(["info", "debug"], .severity)"#,
-							]
-							syntax: "remap_boolean_expression"
-						}
+						type: condition: syntaxes: [{name: "vrl"}]
```
Comment (jszwedko) [inline website/cue/reference/components/transforms/route.cue:45]: I think this is still not right. `datadog_search` should be supported anywhere we have conditions in Vector. I'd suggest we could just leave out passing along which syntaxes are supported and just render them all until we have a component that has a condition field that only supports a subset of them.

Comment (lucperkins) [reply website/cue/reference/components/transforms/route.cue:45]: As per #10257, it looks like it isn't _yet_ the case that `datadog_search` is supported everywhere. But it sounds like @leebenson is going to pull the strings to make that happen. So I'll rework this PR to reflect that and make sure to merge only after those changes have been made.

Comment (jszwedko) [reply website/cue/reference/components/transforms/route.cue:45]: 👍 As noted by https://github.com/vectordotdev/vector/issues/10257#issuecomment-985785321 it is the case that it is supported everywhere already.

Comment (lucperkins) [reply website/cue/reference/components/transforms/route.cue:45]: Ah, derp, I misinterpreted that! Okay, that makes all of this easier.

**Comment by lucperkins** [comment]
@jszwedko Okay, I've updated this. Now, all you need to is `type: condition: {}` to get the full meal deal. I think that should make things much easier for others.

**Review by jszwedko — APPROVED** [review]
Looks good! Thanks for adding the short-hand details.

**Pull request merged**